MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1325,5 +1336,120 @@ void EnggDiffractionPresenter::calcVanadiumWorkspaces(
   vanCurvesWS = ADS.retrieveWS<MatrixWorkspace>(curvesName);
 }
 
+/**
+ * Checks the plot type selected and applies the appropriate
+ * python function to apply during first bank and second bank
+ *
+ * @param outWSName; title of the focused workspace
+ */
+void EnggDiffractionPresenter::plotFocusedWorkspace(std::string outWSName) {
+  const bool plotFocusedWS = m_view->focusedOutWorkspace();
+  int plotType = m_view->currentPlotType();
+  if (plotFocusedWS == true && 0 == plotType) {
+    if (outWSName == "engggui_focusing_output_ws_bank_1")
+      m_view->plotFocusedSpectrum(outWSName);
+    if (outWSName == "engggui_focusing_output_ws_bank_2")
+      m_view->plotReplacingWindow(outWSName);
+  } else if (plotFocusedWS == true && 1 == plotType) {
+    if (outWSName == "engggui_focusing_output_ws_bank_1")
+      m_view->plotFocusedSpectrum(outWSName);
+    if (outWSName == "engggui_focusing_output_ws_bank_2")
+      m_view->plotWaterfallSpectrum(outWSName);
+  } else if (plotFocusedWS == true && 2 == plotType) {
+    m_view->plotFocusedSpectrum(outWSName);
+  }
+}
+
+/**
+ * Convert the generated output files and saves them in
+ * FocusedXYE format
+ *
+ * @param inputWorkspace; title of the focused workspace
+ */
+void EnggDiffractionPresenter::saveFocusedXYE(
+    const std::string inputWorkspace) {
+  auto alg = Algorithm::fromString("SaveFocusedXYE");
+  std::string fullFilename = inputWorkspace + "_FocusedxyeFile.dat";
+  try {
+    g_log.debug() << "Going to save focused output into OpenGenie file: "
+                  << fullFilename << std::endl;
+    alg->initialize();
+    alg->setProperty("InputWorkspace", inputWorkspace);
+
+    std::string filename(fullFilename);
+    alg->setPropertyValue("Filename", filename);
+    alg->execute();
+  } catch (std::runtime_error &re) {
+    g_log.error() << "Error in saving focused XYE file. ",
+        "Could not run the algorithm SaveFocusedXYE succesfully for "
+        "workspace " +
+            inputWorkspace + ". Error description: " + re.what() +
+            " Please check also the log messages for details.";
+    throw;
+  }
+  g_log.notice() << "Saved focused workspace as file: " << fullFilename
+                 << std::endl;
+}
+
+/**
+ * Convert the generated output files and saves them in
+ * GSS format
+ *
+ * @param inputWorkspace; title of the focused workspace
+ */
+void EnggDiffractionPresenter::saveGSS(const std::string inputWorkspace) {
+  auto alg = Algorithm::fromString("SaveGSS");
+  std::string fullFilename = inputWorkspace + "_GssFile.gss";
+  try {
+    g_log.debug() << "Going to save focused output into OpenGenie file: "
+                  << fullFilename << std::endl;
+    alg->initialize();
```

Comment:
Oh yes, this should have the `bank_1`, `bank_2` suffix added to the name. The suffixes that Saurabh had in mind are in the description of this issue: https://github.com/mantidproject/mantid/issues/13783 (bank_, texture_, cropped_)